src/Aspire.Dashboard/Components/Controls/AspireMenu.razor.cs

```diff
@@ -37,11 +41,51 @@ public async Task CloseAsync()
         }
     }
 
-    public async Task OpenAsync(int clientX, int clientY)
+    public async Task OpenAsync(int screenWidth, int screenHeight, int clientX, int clientY)
     {
         if (_menu is { } menu)
         {
-            await menu.OpenAsync(clientX, clientY);
+            // Calculate the position to display the context menu using the cursor position (clientX, clientY)
+            // together with the screen width and height.
+            // The menu may need to be displayed above or left of the cursor to fit in the screen.
+            var left = 0;
+            var right = 0;
+            var top = 0;
+            var bottom = 0;
+
+            if (clientX + menu.HorizontalThreshold > screenWidth)
+            {
+                right = screenWidth - clientX;
+            }
+            else
+            {
+                left = clientX;
+            }
+
+            if (clientY + menu.VerticalThreshold > screenHeight)
+            {
+                bottom = screenHeight - clientY;
+            }
+            else
+            {
+                top = clientY;
+            }
+
+            Style = string.Empty;
```

Comment:
why this line?

Reply:
To reset the style. We don't want to add new positions each time the context menu is opened.

But now that you bring it up, using `StyleBuilder` without argument is a simpler way to do this.